### DISCUSSION TITLE
@walletconnect/modal-react-native not sign-in request and transaction on Metamask for Android system. And Provider problems.

### DISCUSSION THREAD

**Discussion by Lestonz** [WalletConnect/walletconnect-monorepo#3002] [Developer Support]
I am using this library @walletconnect/modal-react-native for my app. In IOS is working Metamask, Trust, Rainbow etc. (Mainnet and Goerli), but Android is not connecting Metamask, that is open but has no sign-in message or transaction message. Rainbow is working for Android. In addition, I am using the web3.js library for my app, `const { provider } = useWalletConnectModal();` is not working well. For example when I asked ETH balance giving an error.

my code for provider;

```
  const { open, provider, isConnected, address } = useWalletConnectModal();
  const web3Provider = useMemo(
      () => new Web3(provider ? new Web3(provider) : undefined), [provider]
  );
    const getMyBalance = async () => {
        const myBalance = await web3.eth.getBalance(address);
        console.log("My Eth Balance: ", myBalance)
    }
```

**Comment by ignaciosantise** [comment]
hey @Lestonz there's an issue in your code, you shouldnt call `new Web3Provider(undefined)`, and also, it would be better to initialize web3 after the session is established

```
const web3Provider = useMemo(() => {
    if (provider && isConnected) {
      return new Web3(provider);
    }
    return null;
  }, [provider, isConnected]);

```

What error are you getting with getBalance? Did you search the issue in web3js [repo](https://github.com/web3/web3.js/issues?q=is%3Aissue+is%3Aopen)? You'll probably have to polyfill things to make it work in react-native

**Reply by Lestonz** [reply]
Hello @ignaciosantise,

I got this error => `**TypeError: process.nextTick is not a function. (In 'process.nextTick(cb.bind(null, null, ret))', 'process.nextTick' is undefined)** `.

When I use Infura Provider works for me. However, WalletConnection's provider is not working. 

In addition, Can you help me with my first question? **Android is not connecting Metamask, that is open but has no sign-in message or transaction message. Rainbow is working for Android.** 

Thanks for helping :)

**Reply by ignaciosantise** [reply]
@Lestonz If it works with Rainbow, then it's a MM specific issue that you should report in their repo. 

Currently MM has an issue related with chains, for example if you are in `Ethereum Main Network` in MetaMask, but you send a request using another chain, the modal won't appear. Can you check if our dapp and MetaMask are in the same network?

And about the `process.nextTick` error, you'll need to polyfill things to make web3js work in RN. [This](https://stackoverflow.com/a/55827235/6562073) might be helpful
